### PR TITLE
[Altair] Update to latest spec v1.1.0-alpha.6

### PR DESCRIPTION
**Pull request by michaelsproul**
## Proposed Changes

* Update our Altair consensus code to v1.1.0-alpha.6
* Add new transition tests, new epoch processing tests, and read metadata file that wasn't read as part of genesis validity tests.
* Update config handling to use the presets and configs defined by the EF spec.

## Additional Info

* Other alpha.6 changes will need to occur in their relevant PRs, e.g. the changes to gossip verification should go in #2321.

### CODE REVIEW

**Comment by michaelsproul** [comment]
Ready for review.

I'm going to experiment now with trying to make the `/config/spec` endpoint backwards compatible with validator clients running Lighthouse v1.4.x and earlier. I initially thought that breaking backwards compatibility would be necessary, but I think we can fix it. For background see `#apis` on Eth R&D: https://discord.com/channels/595666850260713488/710831784991916072/849550587090436126

**Comment by michaelsproul** [comment]
Ok. This is really ready for review now.

I'm satisfied with the compatibility shim. It only adds a few fields manually to `extra_fields`, and can be switched off via `--http-disable-legacy-spec` on the BN (I haven't added an equivalent option to the VC).

In the process of testing I also realised I'd corrupted the testnet configs. They should be fixed now (I've checked that the specs agree between Lighthouse VC v1.3.0 and this branch, for Pyrmont and Prater).

File: common/eth2_network_config/src/lib.rs [hunk 1]
```diff
@@ -260,11 +248,7 @@ mod tests {
             let config = Eth2NetworkConfig::from_hardcoded_net(net)
                 .unwrap_or_else(|_| panic!("{:?}", net.name));
 
-            if net.name == "mainnet"
-                || net.name == "toledo"
-                || net.name == "pyrmont"
-                || net.name == "prater"
-            {
+            if net.name == "mainnet" || net.name == "pyrmont" || net.name == "prater" {
```
Comment (paulhauner) [inline common/eth2_network_config/src/lib.rs]: Perhaps we can drop this `if` statement? IIRC it's a relic from when we had v0.12 "legacy" specs hanging around.

File: testing/ef_tests/tests/tests.rs [hunk 2]
```diff
@@ -33,14 +34,17 @@ fn config_test<E: EthSpec + TypeName>() {
 }
 
 #[test]
+#[should_panic]
```
Comment (paulhauner) [inline testing/ef_tests/tests/tests.rs]: Just a heads-up that there's a `should_panic` are here and on L43.

These will *probably* get refactored once the configs are released, but it seemed like a risk of "deleted the block-comment tags and tests give green tick so they must have passed".

Feel free to ignore.

**Review by paulhauner — APPROVED** [review]
Looks great! I couldn't find anything that needs changing :rocket: I left two small notes, but I'd be happy to merge without them being addressed.

**Comment by michaelsproul** [comment]
Thanks for review! I've addressed all comments.

I'll merge this wholesale into `altair` to simplify the diff on the upstream PRs (they'll recognise the existing commits)

**Pull request merged**